velox/type/DecimalUtil.h

```diff
@@ -147,6 +147,63 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  // Derives from Arrow function DecimalFromString.
+  template <typename TOutput>
+  inline static std::optional<TOutput> rescaleVarchar(
```

Comment:
It would be nice to extract this change into a separate PR with its own tests. 

Does all this code need to be in the .h file? Can some of it be moved into .cpp file to speed up builds?

Reply:
This one is only used in Cast, does it need to be in DecimalUtil.h? Can I move it to `velox/expression/CastExpr-inl.h` and add `velox/expression/CastExpr-inl.cpp`?

I just align with DecimalUtil::rescaleInt.

Reply:
I'll let @majetideepak to comment on that.

Reply:
Hi, @majetideepak Can you take a time to look into this discussion?

Reply:
@jinchengchenghh I am okay with moving decimal cast helper functions to CastExpr* if they are only used there. But since these are template functions, you must put them in a header file right?
Am I missing something?

Reply:
We can put it to cpp file https://stackoverflow.com/questions/115703/storing-c-template-function-definitions-in-a-cpp-file
But put it to .h file maybe more simple, so I will put it in CastExpr-inl.h